lib/logstash/outputs/sqs.rb

```diff
@@ -1,136 +1,163 @@
+#     message lengths) is also 256KiB.
 #
-# See http://aws.amazon.com/iam/ for more details on setting up AWS identities.
+# When operating in batch mode, this plugin will dynamically adjust the size of
+# the batch to ensure that the total payload size does not exceed 256KiB. If,
+# however, any single message exceeds the 256KiB then it will be dropped.
 #
 class LogStash::Outputs::SQS < LogStash::Outputs::Base
-  include LogStash::PluginMixins::AwsConfig
-  include Stud::Buffer
+  include LogStash::PluginMixins::AwsConfig::V2
 
-  config_name "sqs"
+  config_name 'sqs'
+  default :codec, 'json'
 
-  # Name of SQS queue to push messages into. Note that this is just the name of the queue, not the URL or ARN.
-  config :queue, :validate => :string, :required => true
+  concurrency :shared
 
-  # Set to true if you want send messages to SQS in batches with `batch_send`
-  # from the amazon sdk
-  config :batch, :validate => :boolean, :default => true
+  # Set to `true` to send messages to SQS in batches. The size of the batch is
+  # configurable via `batch_events`.
+  config :batch, :validate => :boolean, :default => true, :deprecated => true
 
-  # If `batch` is set to true, the number of events we queue up for a `batch_send`.
+  # The number of events to be sent in each batch. This is only relevant when
+  # `batch` is set to `true`.
   config :batch_events, :validate => :number, :default => 10
 
-  # If `batch` is set to true, the maximum amount of time between `batch_send` commands when there are pending events to flush.
-  config :batch_timeout, :validate => :number, :default => 5
+  config :batch_timeout, :validate => :number, :deprecate => 'This setting no longer has any effect.'
 
-  public
-  def aws_service_endpoint(region)
-    return {
-        :sqs_endpoint => "sqs.#{region}.amazonaws.com"
-    }
-  end
+  # The maximum number of bytes for any message sent to SQS. Messages exceeding
+  # this size will be dropped. See
+  # http://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/limits-messages.html.
+  config :message_max_size, :validate => :bytes, :default => '256KiB'
 
-  public 
-  def register
-    require "aws-sdk"
+  # The name of the SQS queue to push messages into.
+  config :queue, :validate => :string, :required => true
 
-    @sqs = AWS::SQS.new(aws_options_hash)
+  public
+  def register
+    @sqs = Aws::SQS::Client.new(aws_options_hash)
 
-    if @batch
-      if @batch_events > 10
-        raise RuntimeError.new(
-          "AWS only allows a batch_events parameter of 10 or less"
-        )
-      elsif @batch_events <= 1
-        raise RuntimeError.new(
-          "batch_events parameter must be greater than 1 (or its not a batch)"
-        )
-      end
-      buffer_initialize(
-        :max_items => @batch_events,
-        :max_interval => @batch_timeout,
-        :logger => @logger
-      )
-    end
+    if @batch_events > 10
```

Comment:
I'd keep it in. The chance of a user setting it to higher than ten, and then having an issue, is higher than that of Amazon increasing the batch size.